hil/config.py

```diff
@@ -8,13 +8,85 @@
 import ConfigParser
 import logging.handlers
 import importlib
+from schema import Schema, Optional
 import os
 import sys
+from urlparse import urlparse
 
 cfg = ConfigParser.RawConfigParser()
 cfg.optionxform = str
 
 
+def string_is_bool(option):
+    """Check if a string matches ConfigParser's definition of a bool"""
+    return option.lower() in ['true', 'yes', 'on', '1',
```

Comment:
You could be making better use of the library; `Use`, `And`, and `Or` are handy:

```python
return And(Use(option.lower), Or('true', 'yes', 'on', ...))
```

This can be applied in a number of other places.

Reply:
For some reason, having this function like this instead just places whatever is in `option` into my validation schema, causing any config string to pass.

Reply:
It works however if I put the `And(...)` directly into the schema

Reply:
@zenhack how partial are you to using more of these library functions? I've been hacking at it for a while, and returning in this style causes the validation to not happen. From debugging it looks like there's an issue with returning the validation from another function, since it does work when hard-coding into the Schema dictionary.

To use more of the library functions I could instead remove some of the functions (like `string_is_bool`) and write the validation code right into the schema.

Reply:
Oh, that's what I was missing. That definitely works, but it comes down now to how ugly the error messages look:

Current code error example:
```
Error in config file: string_is_bool('Frue') should evaluate to True
```

New suggestion error example: 
```
Error in config file: Or('true', 'yes', 'on', '1', 'false', 'no', 'off', '0') did not validate 'Frue'
'0' does not match 'Frue'
```
I'm not 100% sure why the errors look different now, perhaps because it's a separate validation happening before the main dictionary validation.  I think the 2nd error option might expose a little too much code to the user.

Reply:
I could go either way re: the error message. They're both exposing implementation detail a bit, but the later has the virtue that it actually gives the user a good hint as to what the correct settnigs are.